cooperative_lanechange/src/cooperative_lanechange_node.cpp

```diff
@@ -0,0 +1,664 @@
+    config_.control_plugin_name = declare_parameter<std::string>("control_plugin_name", config_.control_plugin_name);
+    config_.minimum_speed = declare_parameter<double>("minimum_speed", config_.minimum_speed);
+    config_.max_accel = declare_parameter<double>("max_accel", config_.max_accel);
+    config_.minimum_lookahead_distance = declare_parameter<double>("minimum_lookahead_distance", config_.minimum_lookahead_distance);
+    config_.maximum_lookahead_distance = declare_parameter<double>("maximum_lookahead_distance", config_.maximum_lookahead_distance);
+    config_.minimum_lookahead_speed = declare_parameter<double>("minimum_lookahead_speed", config_.minimum_lookahead_speed);
+    config_.maximum_lookahead_speed = declare_parameter<double>("maximum_lookahead_speed", config_.maximum_lookahead_distance);
+    config_.lateral_accel_limit = declare_parameter<double>("lateral_accel_limit", config_.lateral_accel_limit);
+    config_.speed_moving_average_window_size = declare_parameter<int>("speed_moving_average_window_size", config_.speed_moving_average_window_size);
+    config_.curvature_moving_average_window_size = declare_parameter<int>("curvature_moving_average_window_size", config_.curvature_moving_average_window_size);
+    config_.curvature_calc_lookahead_count = declare_parameter<int>("curvature_calc_lookahead_count", config_.curvature_calc_lookahead_count);
+    config_.downsample_ratio = declare_parameter<int>("downsample_ratio", config_.downsample_ratio);
+    config_.destination_range = declare_parameter<double>("destination_range", config_.destination_range);
+    config_.lanechange_time_out = declare_parameter<double>("lanechange_time_out", config_.lanechange_time_out);
+    config_.min_timestep = declare_parameter<double>("min_timestep", config_.min_timestep);
+    config_.starting_downtrack_range = declare_parameter<double>("starting_downtrack_range", config_.starting_downtrack_range);
+    config_.starting_fraction = declare_parameter<double>("starting_fraction", config_.starting_fraction);
+    config_.mid_fraction = declare_parameter<double>("mid_fraction", config_.mid_fraction);
+    config_.min_desired_gap = declare_parameter<double>("min_desired_gap", config_.min_desired_gap);
+    config_.desired_time_gap = declare_parameter<double>("desired_time_gap", config_.desired_time_gap);
+    config_.turn_downsample_ratio = declare_parameter<int>("turn_downsample_ratio", config_.turn_downsample_ratio);
+    config_.curve_resample_step_size = declare_parameter<double>("curve_resample_step_size", config_.curve_resample_step_size);
+    config_.back_distance = declare_parameter<double>("back_distance", config_.back_distance);
+    config_.buffer_ending_downtrack = declare_parameter<double>("buffer_ending_downtrack", config_.buffer_ending_downtrack);
+    config_.vehicle_id = declare_parameter<std::string>("vehicle_id", config_.vehicle_id);
+
+    // Initialize World Model
+    wm_ = get_world_model();
```

Comment:
Probably better to do this in on_configure since its creates subscriptions

Reply:
Good point, I've moved it to on_configure_plugin()